src/Sylius/Bundle/ApiBundle/Doctrine/ORM/QueryExtension/Shop/Order/StateBasedExtension.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Doctrine\ORM\QueryExtension\Shop\Order;
+
+use ApiPlatform\Doctrine\Orm\Extension\QueryCollectionExtensionInterface;
+use ApiPlatform\Doctrine\Orm\Extension\QueryItemExtensionInterface;
+use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Metadata\Operation;
+use Doctrine\ORM\QueryBuilder;
+use Sylius\Bundle\ApiBundle\SectionResolver\ShopApiSection;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionProviderInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+
+final readonly class StateBasedExtension implements QueryCollectionExtensionInterface, QueryItemExtensionInterface
+{
+    /** @param array<string> $nonFilteredCartAllowedOperations */
+    public function __construct(
+        private SectionProviderInterface $sectionProvider,
+        private array $nonFilteredCartAllowedOperations = [],
```

Comment:
I suggest either:
```suggestion
        private array $cartAllowedOperations = [],
        or
        private array $allowedOperations = [],
```
I don't see the value in the `nonFiltered` prefix.